### PR TITLE
Completed Exercise 1

### PR DESCRIPTION
**Pull request by apexcz**
Please make sure you are raising this PR against your own repository and not the original. If it says https://github.com/CorndelWithSoftwire in the search bar right now, you are in the wrong place!

Please also don't forget to submit the PR's url to Aptem afterwards (by clicking on the corresponding exercise component, pasting the url in the textbox that appears and then pressing the finish button).

### CODE REVIEW

File: todo_app/templates/index.html [hunk 1]
```diff
@@ -7,12 +7,65 @@ <h1 class="display-4">To-Do App</h1>
...
+          <tr>
+            <th>#</th>
+            <th>Title</th>
+            <th>Status</th>
+            <th>Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          {% for index in range(0, todos|length) %}
+              {% set todo = todos[index] %}
```
Comment (JackMeadDev) [inline todo_app/templates/index.html]: Nice solution to this! Jinja provides us with an inbuilt way to grab the index from the loop though, so we could equally do the below, which might be a little clearer
```python
{% for todo in todos %}
              <tr>
                <td>{{loop.index}}</td>
                <td>{{todo.title}}</td>
... 
```

File: todo_app/templates/index.html [hunk 2]
```diff
@@ -7,12 +7,65 @@ <h1 class="display-4">To-Do App</h1>
...
+              <tr>
+                <td>{{index + 1}}</td>
+                <td>{{todo.title}}</td>
+                <td>{{todo.status}}</td>
+
+                <td class="d-flex justify-content-center">
+
+                  {% if todo.status != 'Completed' %}
+                    <form action="{{ url_for('mark_completed') }}" method="post">
+                      <input type="hidden" name="todo_item" value='{{ todo|tojson }}'>
```
Comment (JackMeadDev) [inline todo_app/templates/index.html]: Passing through the value as its entire JSON form makes things a little more involved - instead we'd usually just pass through the id & then complete or delete the item matching that id

File: todo_app/app.py [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
 app = Flask(__name__)
 app.config.from_object(Config)
 
 
-@app.route('/')
+@app.route('/', methods=["GET", "POST"])
 def index():
-    return 'Hello World!'
+    error_message = None
+    if request.method == "POST":
```
Comment (JackMeadDev) [inline todo_app/app.py]: The fact we handle this under the same route makes this a little more complex; splitting out adding cards into its own route might make more sense (with the endpoint potentially just returning an error if the problem occurs)

**Review by JackMeadDev — APPROVED** [review]
This looks good, I was able to easily run the app & create/delete/complete your todo items. I've left a few thoughts below, but happy for you to merge this when you're ready

File: todo_app/templates/index.html (hunk 1 above)
Comment (apexcz) [reply todo_app/templates/index.html]: Cheers Jack, done.

File: todo_app/templates/index.html (hunk 2 above)
Comment (apexcz) [reply todo_app/templates/index.html]: Cheers Jack, done.

File: todo_app/app.py (hunk 3 above)
Comment (apexcz) [reply todo_app/app.py]: Cheers Jack, done.

**Pull request merged**